### DISCUSSION TITLE
Windows variables & imagemagick command

### DISCUSSION THREAD

**Discussion by AlanEdd** [ImageMagick/ImageMagick#7034] [Help]
### ImageMagick version

 7.1.1-26 Q8 x64 83eefaf:20240107

### Operating system, version and so on

Windows 10 

### Description

Hey there, I am using a batchfile in Windows to do a longer script. I have condensed down the issue, I know it's something simple and I have been googling, just can't see the error. I am just trying to use windows variables to drop into the imagemagick command line. I believe I have to use %% for %character. Anyway it's just not transferring the variable into the command. This is what I am using:
```
SET /p cropinfo="-crop 100x100+20+20"
ECHO magick -verbose in.jpg %%cropinfo%% out.jpg
```
Any pointers much appreciated.

**Comment by snibgo** [comment]
You shouldn't double the percent signs in `%cropinfo%`.

If you wanted magick to see `%cropinfo%`, you would need to double the percent signs. But you don't want that, you want magick to see the *translation* of the environment variable. The translation is done by the Windows CMD interpreter, before it executes the magick command. To get Windows to translate it, you need single percent signs.

I think your `SET` command doesn't do what you want it to do.
